xarray/backends/cfgrib_.py

```diff
@@ -91,24 +98,24 @@ def get_encoding(self):
 
 
 class CfgribfBackendEntrypoint(BackendEntrypoint):
-    available = module_available("cfgrib")
-
-    def guess_can_open(self, filename_or_obj):
-        try:
+    def guess_can_open(
+        self,
+        filename_or_obj: str | os.PathLike[Any] | BufferedIOBase | AbstractDataStore,
+    ) -> bool:
+        if isinstance(filename_or_obj, (str, os.PathLike)):
             _, ext = os.path.splitext(filename_or_obj)
-        except TypeError:
-            return False
-        return ext in {".grib", ".grib2", ".grb", ".grb2"}
+            return ext in {".grib", ".grib2", ".grb", ".grb2"}
+        return False
 
-    def open_dataset(
+    def open_dataset(  # type: ignore[override]  # allow LSP violation, not supporting **kwargs
```

Comment:
What was the issue here? Was it already there and adding the typing highlighted the issue?

Reply:
Like the comment says, this implementation violates liskov substitution principle. The base class accepts any kwarg using **kwarg but this inherited one only allows certain ones.
A subclass should only extend arguments but not restrict them.

We could also change the base class function signature instead.